components/aws/sagemaker/tests/integration_tests/component_tests/test_groundtruth_component.py

```diff
@@ -110,20 +111,24 @@ def test_groundtruth_labeling_job(
         )
         assert response["LabelingJobStatus"] in ["Stopping", "Stopped"]
     finally:
-        # Check if terminate failed, and stop the labeling job
-        labeling_jobs = sagemaker_utils.list_labeling_jobs_for_workteam(
-            sagemaker_client, workteam_arn
+        print(
+            f"Clean up workteam: {workteam_arn} and GT job: {ground_truth_train_job_name}"
         )
-        if len(labeling_jobs["LabelingJobSummaryList"]) > 0:
-            sagemaker_utils.stop_labeling_job(
-                sagemaker_client, ground_truth_train_job_name
-            )
-
-        # Cleanup the workteam
-        workteams = sagemaker_utils.list_workteams(sagemaker_client)["Workteams"]
-        workteam_names = list(map((lambda x: x["WorkteamName"]), workteams))
-        if workteam_name in workteam_names:
-            sagemaker_utils.delete_workteam(sagemaker_client, workteam_name)
+        if workteam_arn:
+            if ground_truth_train_job_name:
+                # Check if terminate failed, and stop the labeling job
+                labeling_jobs = sagemaker_utils.list_labeling_jobs_for_workteam(
+                    sagemaker_client, workteam_arn
+                )
+                if len(labeling_jobs["LabelingJobSummaryList"]) > 0:
+                    sagemaker_utils.stop_labeling_job(
```

Comment:
does this command fail if the job is already stopped (hopefully not). But if yes, then a better check instead of line 118 would be to ensure that there is a job with the said name that needs to be stopped, similar to the workteam check in line 130.

Reply:
last I checked list_jobs wont return the job if it has already stopped so we should be good

Reply:
then we are ok